bin/check_fasta_insdc.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+
+import argparse
+import gzip
+import hashlib
+
+from itertools import groupby
+
+import requests
+from ebi_eva_common_pyutils.logger import logging_config
+
+import yaml
+from requests import HTTPError
+from retry import retry
+
+REFGET_SERVER = 'https://www.ebi.ac.uk/ena/cram'
+
+logger = logging_config.get_logger(__name__)
+
+
+def open_gzip_if_required(input_file):
+    if input_file.endswith('.gz'):
+        return gzip.open(input_file, 'rt')
+    else:
+        return open(input_file, 'r')
+
+
+def write_result_yaml(output_yaml, results):
+    with open(output_yaml, 'w') as open_yaml:
+        yaml.safe_dump(data=results, stream=open_yaml)
+
+
+def refget_md5_digest(sequence):
+    return hashlib.md5(sequence.upper().encode('utf-8')).hexdigest()
+
+
+def fasta_iter(input_fasta):
+    """
+    Given a fasta file. yield tuples of header, sequence
+    """
+    "first open the file outside "
+    fin = open(input_fasta, 'r')
```

Comment:
I believe it's okay to `yield` from within a `with` block if that's the concern here. Either way we should close the file at the end of the iterator...

Reply:
Added